github/resource_github_actions_runner_group_test.go

```diff
@@ -0,0 +1,274 @@
+			  visibility = "all"
+			}
+    `, randomID)
+
+		check := resource.ComposeTestCheckFunc(
+			resource.TestCheckResourceAttrSet("github_actions_runner_group.test", "name"),
+			resource.TestCheckResourceAttrSet("github_actions_runner_group.test", "visibility"),
+			resource.TestCheckResourceAttr("github_actions_runner_group.test", "visibility", "all"),
+		)
+
+		testCase := func(t *testing.T, mode string) {
+			resource.Test(t, resource.TestCase{
+				PreCheck:  func() { skipUnlessMode(t, mode) },
+				Providers: testAccProviders,
+				Steps: []resource.TestStep{
+					{
+						Config: config,
+						Check:  check,
+					},
+					{
+						ResourceName:      "github_actions_runner_group.test",
+						ImportState:       true,
+						ImportStateVerify: true,
+					},
+				},
+			})
+		}
+
+		t.Run("with an anonymous account", func(t *testing.T) {
+			t.Skip("anonymous account not supported for this operation")
+		})
+
+		t.Run("with an individual account", func(t *testing.T) {
+			t.Skip("individual account not supported for this operation")
+		})
+
+		t.Run("with an organization account", func(t *testing.T) {
+			testCase(t, organization)
+		})
+	})
+
+	// This test is currently skipped due to the perpetual diff in runner group visibility when creating a private runner group.
+	t.Run("imports a private runner group without error", func(t *testing.T) {
+		config := fmt.Sprintf(`
+					resource "github_repository" "test" {
+					  name = "tf-acc-test-%s"
+					}
+
+					resource "github_actions_runner_group" "test" {
+					  name       = github_repository.test.name
+					  visibility = "private"
+					}
+		    `, randomID)
+
+		check := resource.ComposeTestCheckFunc(
+			resource.TestCheckResourceAttrSet("github_actions_runner_group.test", "name"),
+			resource.TestCheckResourceAttrSet("github_actions_runner_group.test", "visibility"),
+		)
+
+		testCase := func(t *testing.T, mode string) {
+			resource.Test(t, resource.TestCase{
+				PreCheck:  func() { skipUnlessMode(t, mode) },
+				Providers: testAccProviders,
+				Steps: []resource.TestStep{
+					{
+						Config: config,
+						Check:  check,
+					},
+					{
+						ResourceName:      "github_actions_runner_group.test",
+						ImportState:       true,
+						ImportStateVerify: true,
+					},
+				},
+			})
+		}
+
+		t.Run("with an anonymous account", func(t *testing.T) {
+			t.Skip("anonymous account not supported for this operation")
+		})
+
+		t.Run("with an individual account", func(t *testing.T) {
+			t.Skip("individual account not supported for this operation")
+		})
+
+		t.Run("with an organization account", func(t *testing.T) {
+			// Note: this test is skipped because when setting visibility 'private', it always fails with:
+			// Step 0 error: After applying this step, the plan was not empty:
+			// visibility:                 "all" => "private"
+			t.Skip("always shows a diff for visibility 'all' => 'private'")
+			testCase(t, organization)
```

Comment:
I wasn't sure if I should leave this test here or take it out entirely.

Reply:
Safe to leave in for this first pass.  We may upgrade our testing suite and revisit in the future 🤞🏾